microprofile/metrics-smallrye/src/main/java/org/wildfly/extension/microprofile/metrics/MicroProfileMetricsSubsystemAdd.java

```diff
@@ -67,8 +69,15 @@ public void execute(DeploymentProcessorTarget processorTarget) {
         List<String> exposedSubsystems = MicroProfileMetricsSubsystemDefinition.EXPOSED_SUBSYSTEMS.unwrap(context, model);
         String prefix = MicroProfileMetricsSubsystemDefinition.PREFIX.resolveModelAttribute(context, model).asStringOrNull();
 
-        MetricsRegistrationService.install(context, exposedSubsystems, prefix);
-        MetricsContextService.install(context, securityEnabled);
+        context.addStep(new OperationStepHandler() {
+            @Override
+            public void execute(OperationContext operationContext, ModelNode modelNode) {
+                MetricsContextService.install(context, securityEnabled);
+                MetricsRegistrationService.install(context, exposedSubsystems, prefix);
+
+            }
+        }, VERIFY);
```

Comment:
Perhaps a comment saying that no one but you is allowed to write code that does this?

j/k

I don't have any better ideas!

Reply:
@bstansberry that's ugly and very hacky :(
`VERIFY` is a special stage and I agree that I should not have to plug my services at that stage to be sure that I'm the "last" runtime service to start...

Another approach would be to use a StabilityMonitor to wait for "some" service stability to ensure all runtime resources are created and only after that install the MetricsRegistrationService. (which should equally ugly and hacky...)